Cython/Compiler/Nodes.py

```diff
@@ -1718,8 +1718,17 @@ def analyse_declarations(self, env):
         if self.items is not None:
             if self.in_pxd and not env.in_cinclude:
                 self.entry.defined_in_pxd = 1
+
+            # For extern enums, we can't reason about their equivalent int values because
+            # we don't know if their definition is complete.
+            is_declared_enum = self.visibility != 'extern'
+
+            next_int_enum_value = 0 if is_declared_enum else None
             for item in self.items:
-                item.analyse_declarations(scope, self.entry)
+                item.analyse_enum_declarations(scope, self.entry, next_int_enum_value)
+                if is_declared_enum:
+                    next_int_enum_value = (
+                        item.entry.enum_int_value or next_int_enum_value) + 1
```

Comment:
I now wonder if this is actually safe. What if users declare their enums in an unusual order, like:
```
cdef enum x:
    d = 4
    a = 0
    b, c
```
Wouldn't we lose the `a = 0` in this case? I think we need to test for `None` here instead of a simple truthiness check. (Probably worth adding a declaration like this as a test.)